### PR TITLE
Feat: support HIP-32, HIP-206, and HIP-222

### LINKED ISSUE #789: HTS Precompile Transaction Record Support
### Problem

HTS precompile transactions will introduce changes to the content returned in a transaction record. 

[HIP-206](https://hips.hedera.com/hip/hip-206)

HIP will be updated with final transaction record details

Hedera Protobuf PR: https://github.com/hashgraph/hedera-protobufs/pull/120

### Solution

Add support for the new fields

### Alternatives

_No response_

### LINKED ISSUE #790: Add ECDSA(secp256k1) key support
### Problem

Hedera will be adding support for ECDSA(secp256k1) keys

[HIP-222](https://hips.hedera.com/hip/hip-222)

Hedera protobuf PR: https://github.com/hashgraph/hedera-protobufs/pull/115

### Solution

Add SDK support for this key type

### Alternatives

_No response_

### LINKED ISSUE #791: Add auto account creation support
### Problem

Hedera is adding an alternative way to create accounts

[HIP-32](https://hips.hedera.com/hip/hip-32)

The HIP needs to be updated with the latest specification

Hedera protobufs PR: https://github.com/hashgraph/hedera-protobufs/pull/119

### Solution

Add support for auto account creation

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Sean-Tedrow-LB**
**Description**:

Add support for [HIP-32](https://hips.hedera.com/hip/hip-32)
Add support for [HIP-206](https://hips.hedera.com/hip/hip-206)
Add support for [HIP-222](https://hips.hedera.com/hip/hip-222)

**Related issue(s)**:

Fixes #791
Fixes #789
Fixes #790

**Checklist**

- [X] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionRecord.java [hunk 1]
```diff
@@ -221,6 +241,10 @@ public static TransactionRecord fromBytes(byte[] bytes) throws InvalidProtocolBu
             transactionRecord.addAutomaticTokenAssociations(tokenAssociation.toProtobuf());
         }
 
+        if (aliasKey != null) {
+            transactionRecord.setAlias(ByteString.copyFrom(aliasKey.toBytes()));
```
Comment (danielakhterov) [inline sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionRecord.java:245]:
```suggestion
            transactionRecord.setAlias(aliasKey.toProtobufKey().toByteString());
```

File: sdk/src/main/java/com/hedera/hashgraph/sdk/AccountId.java [hunk 2]
```diff
@@ -103,16 +154,24 @@ public String getChecksum() {
...
+            return "" + shard + "." + realm + "." + aliasKey.toStringDER();
+        } else {
+            return EntityIdHelper.toString(shard, realm, num);
+        }
     }
 
     public String toStringWithChecksum(Client client) {
-        return EntityIdHelper.toStringWithChecksum(shard, realm, num, client, checksum);
+        if (aliasKey != null) {
+            return "" + shard + "." + realm + "." + aliasKey.toStringDER();
```
Comment (danielakhterov) [inline sdk/src/main/java/com/hedera/hashgraph/sdk/AccountId.java:166]: Should probably error here instead of serializing just like `toString()`.

**Review by danielakhterov — CHANGES_REQUESTED** [review]

File: sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionReceiptQuery.java [hunk 3]
```diff
@@ -45,6 +49,24 @@ public TransactionReceiptQuery setTransactionId(TransactionId transactionId) {
         return this;
     }
 
+    public boolean getIncludeChildren() {
+        return includeChildren;
+    }
+
+    public TransactionReceiptQuery setIncludeChildren(boolean value) {
```
Comment (SimiHunjan) [inline sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionReceiptQuery.java:56]: add comments for `setIncludeChildren`, `setIncludeDuplicates`, etc

File: sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionRecord.java [hunk 4]
```diff
@@ -98,7 +103,9 @@ private TransactionRecord(
         Map<TokenId, List<TokenNftTransfer>> tokenNftTransfers,
         @Nullable ScheduleId scheduleRef,
         List<AssessedCustomFee> assessedCustomFees,
-        List<TokenAssociation> automaticTokenAssociations
+        List<TokenAssociation> automaticTokenAssociations,
+        @Nullable PublicKey aliasKey,
+        List<TransactionRecord> children
```
Comment (SimiHunjan) [inline sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionRecord.java:108]: Is this representing the parent consensus timestamp? I didn't see a children field being returned in the protobuf here: https://github.com/hashgraph/hedera-protobufs/blob/v0.21.0/services/transaction_record.proto

Comment (SimiHunjan) [reply sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionRecord.java:108]: understand why children is there now. Please add parent consensus timestamp.

**Review by danielakhterov — APPROVED** [review]

**Pull request merged**